### PR TITLE
feature: add NFT Standard Cadence Project

### PR DESCRIPTION
**Pull request by FelipeRibeiroLabs**
Flow NFT Scaffold contains the NFT-Standards folder structure you will need to develop your NFT project. I provided some examples of contracts and transactions you can modify and take as an example.

______

For contributor use:

- [x] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-cli/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

**Comment by sideninja** [comment]
@FelipeRibeiroLabs very nice work! let me include some other people who are expert on Cadence cc @joshuahannan @aishairzay 

This PR will be blocked by the issue: https://github.com/onflow/flow-cli/issues/797 but I plan to work on it next days.

File: scaffolds.json [hunk 1]
```diff
@@ -4,5 +4,11 @@
     "repo": "https://github.com/sideninja/flow-basic-scaffold.git",
     "branch": "main",
     "description": "Scaffold contains required folder structure as well as some example Cadence code."
+  },
+  {
+    "name": "NFT Standard Cadence Project",
+    "repo": "https://github.com/FelipeRibeiroLabs/flow-nft-scaffold.git",
+    "branch": "main",
+    "description": "Flow NFT Scaffold contains the NFT-Standards folder structure you will need to develop your NFT project. I provided some examples of contracts and transactions you can modify and take as an example."
```
Comment (sideninja) [inline scaffolds.json:12]: description should be ideally shorted, aprox length to the one above.

**Comment by joshuahannan** [comment]
@sideninja Can you give me a bit of context on what this is for? I just see a link to an external github repo

**Comment by sideninja** [comment]
> @sideninja Can you give me a bit of context on what this is for? I just see a link to an external github repo

Sorry @joshuahannan The new CLI contains a `setup` command, which setups your project to at least contain all the standard folders (contracts, scripts etc) but it also supports creating a project from a scaffold, which is a list of other repos that contain content that deemed to be useful for users starting on different project. So they can choose a scaffold based on what they are trying to do and on which are available in the first place. This is supposed to grow as community keeps adding them.

**Comment by joshuahannan** [comment]
For a scaffold this important, I would almost rather it live in the onflow domain so it can be more easy to monitor and update whenever we make updates. @FelipeRibeiroLabs would you be willing to help create that with us? We can create the repo for you and give you write permissions so you can port your scaffold to it. What do you think?

**Comment by FelipeRibeiroLabs** [comment]
> For a scaffold this important, I would almost rather it live in the onflow domain so it can be more easy to monitor and update whenever we make updates. @FelipeRibeiroLabs would you be willing to help create that with us? We can create the repo for you and give you write permissions so you can port your scaffold to it. What do you think?

Yes, sure! Let me know where to post it! I also welcome feedback and suggestions. I am working on adding new tests with the Cadence Testing Library right now!

**Comment by sideninja** [comment]
@joshuahannan any news on this? Btw we are controlling which version of scaffold is included even if it external repo, which I believe would be a good way to scale this when the community is involved. 

@FelipeRibeiroLabs can you make sure the flow.json update reflects the latest changes in schema: https://github.com/onflow/flow-cli/blob/master/scaffolds.json There's added commit hash which fixes it to a specific version.

**Comment by joshuahannan** [comment]
@FelipeRibeiroLabs Thank you again for offering to help! After looking at yours a little bit more, it looks very similar to what we already have in the [flow-nft repo](https://github.com/onflow/flow-nft). The flow NFT repo is not compatible with scaffolds yet though, but we would like it to be! Do you think you'd be able to help us make that happen? I'm happy to talk with you to answer any questions and we'll make sure to give you plenty of credit for contributing! 😃

**Comment by FelipeRibeiroLabs** [comment]
> @FelipeRibeiroLabs Thank you again for offering to help! After looking at yours a little bit more, it looks very similar to what we already have in the [flow-nft repo](https://github.com/onflow/flow-nft). The flow NFT repo is not compatible with scaffolds yet though, but we would like it to be! Do you think you'd be able to help us make that happen? I'm happy to talk with you to answer any questions and we'll make sure to give you plenty of credit for contributing! 😃

But surely! That is the idea of this scaffolding! I am available to help make it possible. What changes to this proposal I have created do you think are necessary? Show me the way that works for you.

**Comment by joshuahannan** [comment]
I think we would just need to update the repo structure so that it conforms with the expected scaffold structure, and then make the necessary changes to the imports and flow.json, but you probably know better than I do what is needed for a scaffold

**Comment by sideninja** [comment]
Closing this in according with the comments

**Pull request closed without merging**